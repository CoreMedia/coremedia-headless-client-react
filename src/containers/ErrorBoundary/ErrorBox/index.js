// @flow
import { TeaserBox } from '../../../components/basic/Box';
import { getColors } from '../../../styles/themes/utils';

const ErrorBox = TeaserBox.extend`
  border: 2px solid ${props => getColors(props.theme.colors, 'red').bgsolid};
`;

ErrorBox.displayName = 'ErrorBox';

ErrorBox.defaultProps = {
  theme: {
    colors: {
      red: {
        bgsolid: 'rgba(221, 52, 40, 1)',
      },
    },
  },
};

export default ErrorBox;
