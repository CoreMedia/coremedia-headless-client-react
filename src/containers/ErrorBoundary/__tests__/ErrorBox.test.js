// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ErrorBox from '../ErrorBox';

describe('ErrorBox Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<ErrorBox>Content</ErrorBox>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
