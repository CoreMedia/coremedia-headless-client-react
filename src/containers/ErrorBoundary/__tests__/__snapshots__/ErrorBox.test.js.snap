// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox Component should render correctly 1`] = `
.c0 {
  position: relative;
  border: 0;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  border: 2px solid rgba(221,52,40,1);
}

.c0 > * {
  box-sizing: border-box;
}

<div
  className="c0"
>
  Content
</div>
`;
